language/libra-tools/transaction-replay/src/lib.rs

```diff
@@ -132,13 +133,14 @@ impl LibraDebugger {
         self.debugger.get_version_by_account_sequence(account, seq)
     }
 
-    pub fn run_session_at_version<F>(&self, version: Version, f: F) -> Result<ChangeSet>
+    pub fn run_session_at_version<F>(&self, version: Version, override_changeset: Option<MoveChanges>, f: F) -> Result<ChangeSet>
     where
-        F: FnOnce(&mut Session<RemoteStorage<DebuggerStateView>>) -> VMResult<()>,
+        F: FnOnce(&mut Session<DeltaStorage<RemoteStorage<DebuggerStateView>>>) -> VMResult<()>,
     {
         let move_vm = MoveVM::new();
         let state_view = DebuggerStateView::new(&*self.debugger, version);
-        let remote_storage = RemoteStorage::new(&state_view);
+        let state_view_storage = RemoteStorage::new(&state_view);
+        let remote_storage = DeltaStorage::new(&state_view_storage, override_changeset.unwrap_or_else(|| MoveChanges::new()));
```

Comment:
Might make sense to use `RemoteStorage` if `override_changeset` is `None`?

Reply:
It might break the typing... The type for the function is `Session<DeltaStorage<RemoteStorage<DebuggerStateView>>> -> VMResult<()>`, so it has to be `DeltaStorage` unless we use some dynamic dispatching over there, which I'm not sure how to do that.